libmscore/spanner.cpp

```diff
@@ -596,13 +597,35 @@ void Spanner::computeEndElement()
                         setTrack2(track());
                   if (ticks().isZero() && isTextLine() && parent())   // special case palette
                         setTicks(score()->lastSegment()->tick() - _tick);
-                  // find last cr on this staff that ends before tick2
 
-                  _endElement = score()->findCRinStaff(tick2(), track2() / VOICES);
+                  if (isLyricsLine()) {
+                        // lyrics ticks should already indicate the segment we want
+                        // except for TEMP_MELISMA_TICKS case
+                        Lyrics* l = toLyricsLine(this)->lyrics();
+                        Fraction tick = (l->ticks().ticks() == Lyrics::TEMP_MELISMA_TICKS) ? l->tick() : l->endTick();
```

Comment:
To make it more clear, was the issue that `LyricsLine::tick2()` returned a wrong tick value or was it `findCRinStaff` that returned a chord from a wrong segment?

Reply:
Well, findCRinStaff() does what it supposed to do, it's the right thing for most line types, it's just the wrong thing for melisma lines which have their own unique rules.  There is no value we can pass that function that will produce what we want, so no way to improve the result just by altering LyricsLine::tick2().  Although now that you mention it, it probably would be better to use that value here instead of Lyrics::tick2().  They aren't necessarily the same, but probably they should be.  I will take another look at this.